initrd/bin/kexec-unseal-key

```diff
@@ -25,22 +25,23 @@ DEBUG "CONFIG_TPM2_TOOLS: $CONFIG_TPM2_TOOLS"
 DEBUG "Show PCRs"
 DEBUG "$(pcrs)"
 
+failed=0
 for tries in 1 2 3; do
-	read -s -p "Enter LUKS Disk Unlock Key passphrase (blank to abort): " tpm_password
+	read -s -p "Enter LUKS LUKS TPM Disk Unlock Key passphrase (blank to abort): " tpm_password
 	echo
 	if [ -z "$tpm_password" ]; then
 		die "Aborting unseal disk encryption key"
 	fi
 
 	DO_WITH_DEBUG --mask-position 6 \
 		tpmr unseal "$TPM_INDEX" "0,1,2,3,4,5,6,7" "$TPM_SIZE" \
-		"$key_file" "$tpm_password"
+		"$key_file" "$tpm_password" || failed=1
 
-	if [ "$?" -eq 0 ]; then
+	if [ "$failed" -eq 0 ]; then
 		exit 0
 	fi
```

Comment:
It looks to me like if you enter an incorrect passphrase once, it won't exit correctly if you get it right on try 2, because `failed` is never reset to 0.  I didn't try it though, did you try it?

A better structure would be to test the command success directly:

```
if DO_WITH_DEBUG --mask-position 6 tpmr unseal <yada yada>; then
        exit 0
fi
```